### PR TITLE
Feature/gse 1212 reusable test steps

### PR DESCRIPTION
**Pull request by nafonso**
#### Description
Adding of reusable Steps based on research and feedback collected

### CODE REVIEW

File: SpatialGDK/SpatialGDK.uplugin [hunk 1]
```diff
@@ -74,6 +74,15 @@
         "Mac"
       ]
     },
+    {
```
Comment (nafonso) [inline SpatialGDK/SpatialGDK.uplugin:77]: I'm not sure how this is creeping up, not the first time. Also happened with Vicky.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp [hunk 2]
```diff
@@ -318,9 +318,22 @@ void ASpatialFunctionalTest::AddStepBlueprint(const FString& StepName, const FWo
...
 {
-	StepDefinitions.Add(StepDefinition);
+	FSpatialFunctionalTestStepDefinition StepDefinitionCopy = StepDefinition;
+
+	StepDefinitionCopy.Workers.Add(Worker);
+
+	StepDefinitions.Add(StepDefinitionCopy);
+}
+
+void ASpatialFunctionalTest::AddStepFromDefinitionMulti(const FSpatialFunctionalTestStepDefinition& StepDefinition, TArray<FWorkerDefinition> Workers)
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:330]: Is this meant to be passed by value? I guess we would just copy it anyway... then is there a reason for the above one to be passed by reference?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h [hunk 3]
```diff
@@ -92,11 +92,16 @@ class SPATIALGDKFUNCTIONALTESTS_API ASpatialFunctionalTest : public AFunctionalT
...
+	UFUNCTION(BlueprintCallable, Category = "Spatial Functional Test", meta = (DisplayName = "Add Step", AutoCreateRefTerm = "IsReadyEvent,StartEvent,TickEvent", ToolTip = "Adds a Test Step. Check GetAllWorkers(), GetAllServerWorkers() and GetAllClientWorkers() for convenience.\n\nIf you split the Worker pin you can define if you want to run on Server, Client or All.\n\nWorker Ids start from 1.\nIf you pass 0 it will run on all the Servers / Clients (there's also a convenience function GetAllWorkersId())\n\nIf you choose WorkerType 'All' it runs on all Servers and Clients (hence WorkerId is ignored).\n\nKeep in mind you can split the Worker pin for convenience."))
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h:95]: Hmm, you really want to split the Worker pin :D

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h [hunk 4]
```diff
@@ -92,11 +92,16 @@ class SPATIALGDKFUNCTIONALTESTS_API ASpatialFunctionalTest : public AFunctionalT
...
 	void AddStepBlueprint(const FString& StepName, const FWorkerDefinition& Worker, const FStepIsReadyDelegate& IsReadyEvent, const FStepStartDelegate& StartEvent, const FStepTickDelegate& TickEvent, float StepTimeLimit = 0.0f);
 
-	UFUNCTION(BlueprintCallable, Category = "Spatial Functional Test")
-	void AddGenericStep(const FSpatialFunctionalTestStepDefinition& StepDefinition);
+	// Add Steps for Blueprints and C++
+
+	UFUNCTION(BlueprintCallable, Category = "Spatial Functional Test", meta = (ToolTip = "Adds a Step from a Definition. Allows you to define a Step and add it / re-use it multiple times.\n\nKeep in mind you can split the Worker pin for convenience."))
+	void AddStepFromDefinition(const FSpatialFunctionalTestStepDefinition& StepDefinition, const FWorkerDefinition& Worker);
+
+	UFUNCTION(BlueprintCallable, Category = "Spatial Functional Test", meta = (ToolTip = "Adds a Step from a Definition. Allows you to define a Step and add it / re-use it multiple times.\n\nKeep in mind you can split the Worker pin for convenience.\nIt is a more extensible version of AddStepFromDefinition(), where you can pass array with multiple specific Workers."))
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h:103]:
```suggestion
	UFUNCTION(BlueprintCallable, Category = "Spatial Functional Test", meta = (ToolTip = "Adds a Step from a Definition. Allows you to define a Step and add it / re-use it multiple times.\n\nKeep in mind you can split the Worker pin for convenience.\nIt is a more extensible version of AddStepFromDefinition(), where you can pass an array with multiple specific Workers."))
```
Nit

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h (hunk 3 above)
Comment (nafonso) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h:95]: It is very convenient, you have one less node :D

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp (hunk 2 above)
Comment (nafonso) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp]: My bad, corrected

**Pull request merged**